### ISSUE TITLE
Bug related to indexed parameters in events

### ISSUE THREAD

**Issue by cicr99** [NethermindEth/warp#890]
Semantic tests related to indexed parameters in events are failing:
- `tests/behaviour/solidity/test/libsolidity/semanticTests/events/event_constructor.sol` -> Fails at cairo compilation
- `tests/behaviour/solidity/test/libsolidity/semanticTests/userDefinedValueType/erc20.sol` -> Fails at deployment
- `tests/behaviour/solidity/test/libsolidity/semanticTests/various/erc20.sol` -> Fails at deployment

**Comment by rjnrohit** [comment]
The first one fails because there is a bug in the `BuiltinHandler` pass #892 . 

The transpilation of the following contract produces invalid cairo code
```solidity
contract ClientReceipt {
    function Deposit(address _from, bytes32 _id, uint _value) public {
	}
    constructor() {
        Deposit(msg.sender, bytes32("abc"), 7);
    }
}
```

**Comment by rjnrohit** [comment]
The other two are failing because the following contract don't deploy 

```solidity
contract ERC20 {
    constructor() {
        _mint(msg.sender);
    }
    function _mint(address account) internal {
        require(account != address(0), "ERC20: mint to the zero address");
    }
}
```

Or possibly because behaviour tests uses by default caller address as `0` , due to this deployments are failing

**Comment by rjnrohit** [comment]
So technically these errors are not related to indexed parameters @cicr99  😅

**Comment by cicr99** [comment]
Yes, I thought that could be possible. These tests were commented on the whitelist because "indexed events" was an unsupported feature. Once I uncommented them, they failed, so actually the issue is for us to investigate the reasons why they did. Thanks for looking into it.

**Comment by rjnrohit** [comment]
Now, the first issue is addressed #900 , the other remaining two can fixed if we use `starknet-devnet` as execution platform for behaviour/semantic tests, for that I've created an issue #901 . Using `starknet-devent` we can pass an account address , in the current situation behv/semantic tests by default uses caller address as `0` during deployment. 

@cicr99  So, you can close this issue. 

PS: I accidentally closed it, but it's up to you now @cicr99

**Comment by swapnilraj** [comment]
Thanks for triage @RohitRanjangit!
